src/sqlite_db.cpp

```diff
@@ -48,8 +51,12 @@
 #include <shared_mutex>
 #include <string>
 
+extern "C" {
+int miopen_sqlite3_memvfs_init(sqlite3* db, char** pzErrMsg, const sqlite3_api_routines* pApi);
+}
 namespace miopen {
 
+template <bool in_mem>
```

Comment:
I dont see why this needs to be templated. The implementation of `sqlite::impl<in_mem>` will always be the same value withing a build of miopen, so why cant the constructor just do `#if MIOPEN_EMBED_DB` inside the implementation?

Reply:
It can, I did not want to clutter the constructor with `#ifdef`s. I will update it to not use templating.

Reply:
Fixed